conda_forge_tick/migrators/nvtools.py

```diff
@@ -0,0 +1,244 @@
+import copy
+import logging
+import os.path
+from typing import Any
+
+import conda_forge_tick.migrators
+from conda_forge_tick.contexts import ClonedFeedstockContext
+from conda_forge_tick.migrators_types import AttrsTypedDict, MigrationUidTypedDict
+from conda_forge_tick.utils import (
+    get_bot_run_url,
+    yaml_safe_dump,
+    yaml_safe_load,
+)
+
+
+def _file_contains(filename, string):
+    with open(filename) as f:
+        return string in f.read()
+
+
+def _insert_requirements_build(meta):
+    insert_before: bool = True
+    before: list[str] = []
+    after: list[str] = []
+    requirements_found = False
+    build_found = False
+    with open(meta) as f:
+        for line in f:
+            if build_found:
+                insert_before = False
+            elif line.startswith("requirements"):
+                requirements_found = True
+            elif requirements_found and line.lstrip().startswith("build"):
+                build_found = True
+
+            if insert_before:
+                before.append(line)
+            else:
+                after.append(line)
+
+    if not (build_found or requirements_found):
+        return
```

Comment:
So is the idea of this check that we are skipping metapackages? Or are there other cases this would be used for?

May help to have a comment above clarifying intent

Reply:
If we didn't find the requirements.build section, then we cannot add our new dependency to the file.